cluster/gce/gci/configure-helper.sh

```diff
@@ -2820,6 +2824,22 @@ EOF
   systemctl restart containerd
 }
 
+function create-sidecar-config {
+  cat >> "/etc/srv/kubernetes/cri_auth_config.yaml" << EOF
+kind: CredentialProviderConfig
+apiVersion: kubelet.config.k8s.io/v1alpha1
+providers:
+  - name: auth-provider-gcp
+    apiVersion: credentialprovider.kubelet.k8s.io/v1alpha1
+    matchImages:
+    - "container.cloud.google.com"
+    - "gcr.io"
+    - "*.gcr.io"
+    - "*.pkg.dev"
+    defaultCacheDuration: 1m
```

Comment:
Would be nice if the duration was set by an env variable which defaults to "1m".

Reply:
I've added a `KUBE_SIDECAR_CACHE_DURATION` environment variable now - `defaultCacheDuration` is still there, but according to the docs on [credential provider responses](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubelet/pkg/apis/credentialprovider/types.go#L71) it should be overridden if the plugin explicitly returns a different cache duration. So the effect should be that the 1m duration is used if the environment variable is not set, but if it is, then the environment variable is used instead.